python/graphstorm/distiller/distiller.py

```diff
@@ -0,0 +1,351 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Distill wrapper.
+"""
+
+import torch as th
+from torch.nn.parallel import DistributedDataParallel
+import os
+import json
+
+from ..model.gnn_distill import GSDistilledModel
+from ..dataloading import DataloaderGenerator, DataManager
+
+class GSdistiller():
+    """ GNN distiller.
+
+    Parameters
+    ----------
+    rank : int
+        The rank.
+        Configs for GNN distillation
+    """
+    def __init__(self, rank):
+        self._rank = rank
```

Comment:
we decide to remove rank from the trainer argument. GraphStorm can get the rank by calling `get_rank()`

Reply:
I removed self.rank by new commit though.